torchinductor/compile_fx.py

```diff
@@ -76,11 +80,18 @@ def compile_fx_inner(
     cudagraphs=None,
     num_fixed=0,
     is_backward=False,
+    graph_id=None,
 ):
-    if count_calls(gm.graph) == 0:
+    if dynamo_utils.count_calls(gm.graph) == 0:
         return gm
 
-    log.info("Compiling %s graph", "BACKWARDS" if is_backward else "FORWARDS")
+    _step_logger()(
+        logging.INFO,
+        "torchinductor compiling "
+        f"{'BACKWARDS' if is_backward else 'FORWARDS'} "
+        f"graph {graph_id}",
```

Comment:
Sorry for late request - it would be great if we can should total # of graphs here as well (e.g. something like "graph 1 of 10"), which will be like a progress bar in the user's mental model and helps with understanding compilation progress.

Reply:
I'm not sure this feature is possible to implement, since the problem of "determining ahead-of-time how many graphs a program, given input, will produce" seems like an undecidable problem.

Reply:
Curious why is it undecidable? I thought the # graphs a program given input will produce is deterministic (or do you mean it is not deterministic?)

Reply:
The number of graphs a program will produce will depend on the input (including possibly randomness) as well, for example, we can't decide without input if the following program will graph break:
```python
if x.sum() > 0:
  print("break!")
else:
  pass
```

My intuition is that counting the ahead of time number of graphs is as challenging as determining if a program graph breaks at all, which is undecidable due to Rice's theorem (the problem of deciding if a program contains a non-trivial semantic property is undecidable).